transport/http_json/multipart_getter.go

```diff
@@ -0,0 +1,71 @@
+package http_json
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const (
+	multipartMaxMemorySize = 1024 * 1024 * 512
+)
+
+type MultipartGetter struct {
+	ParametersGetter
+}
+
+func NewMultipartGetter(req *http.Request) (*MultipartGetter, error) {
+	if req == nil {
+		return nil, fmt.Errorf("request is empty")
+	}
+
+	return &MultipartGetter{
+		ParametersGetter: ParametersGetter{
+			Req: req,
+		},
+	}, nil
+}
+
+// Add upload files to Params like a string
+// params["{inputName}"] = string(file)
+// params["filename_{inputName}"] = string(filename)
+//
+// Example:
+// type MyData struct {
+//     File     string `key:"my_file"`
+//     Filename string `key:"filename_my_file"`
+// }
+func (g *MultipartGetter) Parse() error {
+	if err := g.Req.ParseMultipartForm(multipartMaxMemorySize); err != nil {
+		return fmt.Errorf("ParseMultipartForm %v", err)
+	}
+
+	if err := g.parseForm(); err != nil {
+		return fmt.Errorf("parseForm %v", err)
+	}
+
+	for key, files := range g.Req.MultipartForm.File {
+		if len(files) == 0 {
+			continue
+		}
+
+		file := files[0]
+
+		f, err := file.Open()
```

Comment:
I think, you should return file handler, and developers will be able to process data partially.